include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -774,6 +776,77 @@ __group_scan_fits_in_slm(const sycl::queue& __queue, ::std::size_t __n, ::std::s
     return (__n <= __single_group_upper_limit && __max_slm_size >= __req_slm_size);
 }
 
+template <typename _ValueType, typename _UnaryOp>
+struct __gen_transform_input
+{
+    using __out_value_type = std::decay_t<decltype(::std::declval<_UnaryOp>()(::std::declval<_ValueType>()))>;
```

Comment:
Could this be rewritten with `invoke_result`? Something like:
```suggestion
    using __out_value_type = typename std::invoke_result<_UnaryOp, _ValueType>::type;
```

Reply:
yep thats better.  applied.